### PR TITLE
feat(ironfish,rust,rust-nodejs): Return descriptions from unsigned transaction

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

* Adds getter methods to the unsigned transaction struct
* Adds a `descriptions` method on the Napi layer to fetch parts of an unsigned transaction
* Adds a method to the JS primitive to fetch descriptions

## Testing Plan

Unit test

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/primitives/unsignedTransaction.ts [hunk 1]
```diff
@@ -55,4 +61,40 @@ export class UnsignedTransaction {
...
+
+      const receivedNote = note.decryptNoteForOwner(incomingViewKey)
+      if (receivedNote) {
+        receivedNotes.push(receivedNote)
+      }
+
+      const spentNote = note.decryptNoteForSpender(outgoingViewKey)
+      if (spentNote) {
+        spentNotes.push(spentNote)
+      }
```
Comment (hughy) [inline ironfish/src/primitives/unsignedTransaction.ts:90]: Wondering if we'll want/need to handle notes showing up in both received and spent when we render

File: ironfish/src/primitives/unsignedTransaction.ts [hunk 2]
```diff
@@ -55,4 +61,40 @@ export class UnsignedTransaction {
 
     return result
   }
+
+  descriptions(
+    incomingViewKey: string,
+    outgoingViewKey: string,
+  ): {
+    receivedNotes: Note[]
+    spentNotes: Note[]
```
Comment (hughy) [inline ironfish/src/primitives/unsignedTransaction.ts:70]: nit - should this be `sent` instead of `spent`?

**Review by hughy — APPROVED** [review]
This looks to me like a good way of getting the information that we need from the existing fields and data on an UnsignedTransaction

File: ironfish/src/primitives/unsignedTransaction.ts (hunk 1 above)
Comment (rohanjadvani) [reply ironfish/src/primitives/unsignedTransaction.ts:90]: Yeah. I thought I'd just return everything here for now and then the caller can decide what to do.

**Pull request merged**